dbt/include/spark/macros/materializations/table.sql

```diff
@@ -14,16 +14,22 @@
   -- setup: if the target relation already exists, drop it
   -- in case if the existing and future table is delta or iceberg, we want to do a
   -- create or replace table instead of dropping, so we don't have the table unavailable
-  {% if old_relation and not (old_relation.is_delta and config.get('file_format', validator=validation.any[basestring]) == 'delta') -%}
-    {{ adapter.drop_relation(old_relation) }}
-  {%- endif %}
-
-  {% if old_relation and not (old_relation.is_iceberg and config.get('file_format', validator=validation.any[basestring]) == 'iceberg') -%}
-    {{ adapter.drop_relation(old_relation) }}
-  {%- endif %}
+  {% if old_relation is not none %}
+    {% set is_delta = (old_relation.is_delta and config.get('file_format', validator=validation.any[basestring]) == 'delta') %}
+    {% set is_iceberg = (old_relation.is_iceberg and config.get('file_format', validator=validation.any[basestring]) == 'iceberg') %}
+    {% set old_relation_type = old_relation.type %}
+  {% else %}
+    {% set is_delta = false %}
+    {% set is_iceberg = false %}
+    {% set old_relation_type = target_relation.type %}
+  {% endif %}
+
+  {% if not is_delta and not is_iceberg %}
```

Comment:
Why would we do this only when the table format is not delta or iceberg?

Reply:
That's to mimic the logic that was already there. Judging from the comment, we don't want to drop the table if it's delta or iceberg; so I make sure that it's not that.

Reply:
Yeah I don't know why we'd even do that. This drop behavior is obscure. Maybe it has something to do with table truncation options that are un/available?

I can appreciate you might not know either (was hoping you might tho)